x/stakeibc/keeper/msg_server_restore_interchain_account_test.go

```diff
@@ -0,0 +1,66 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	_ "github.com/stretchr/testify/suite"
+
+	stakeibc "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+type RestoreInterchainAccountTestCase struct {
+	validMsg stakeibc.MsgRestoreInterchainAccount
+}
+
+func (s *KeeperTestSuite) SetupRestoreInterchainAccount() RestoreInterchainAccountTestCase {
+	s.CreateTransferChannel(HostChainId)
+
+	hostZone := stakeibc.HostZone{
+		ChainId:      HostChainId,
+		HostDenom:    Atom,
+		Bech32Prefix: GaiaPrefix,
+		ConnectionId: ibctesting.FirstConnectionID,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+
+	defaultMsg := stakeibc.MsgRestoreInterchainAccount{
+		Creator:     "creatoraddress",
+		ChainId:     HostChainId,
+		AccountType: stakeibc.ICAAccountType_DELEGATION,
+	}
+
+	return RestoreInterchainAccountTestCase{
+		validMsg: defaultMsg,
+	}
+}
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_Success() {
+	tc := s.SetupRestoreInterchainAccount()
+	msg := tc.validMsg
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
+	s.Require().NoError(err, "registered ica account successfully")
+}
+
+func (s *KeeperTestSuite) TestRestoreInterchainAccount_FailsForIncorrectHostZone() {
+	tc := s.SetupRestoreInterchainAccount()
+	msg := tc.validMsg
+	msg.ChainId = "incorrectchainid"
+	_, err := s.GetMsgServer().RestoreInterchainAccount(sdk.WrapSDKContext(s.Ctx()), &msg)
+	s.Require().Error(err, "registered ica account fails for incorrect host zone")
```

Comment:
nit: can we check the error is the expected error, rather than just checking whether it's _any_ error? will help us reason about and notice changes to the implementations on various host zones we test with

Reply:
nit: use `ErrorEqual`

Reply:
+1 on this. Imo it's even a little better to check the exact error with `s.Require().EqualError(err, "error string")` which lets us keep track of the full error messages after all the wrapping (and identifies if we're wrapping in an odd way)